mm2src/lp_stats.rs

```diff
@@ -0,0 +1,285 @@
+
+            let peers_names = match select_peers_names(&ctx) {
+                Ok(n) => n,
+                Err(e) => {
+                    log::error!("Error selecting peers names from db: {}", e);
+                    Timer::sleep(10.).await;
+                    continue;
+                },
+            };
+
+            let timestamp = now_ms() / 1000;
+            let get_versions_res = match request_addresses::<String>(
+                ctx.clone(),
+                P2PRequest::NetworkInfo(NetworkInfoRequest::GetMm2Version),
+                peers_addresses,
+            )
+            .await
+            {
+                Ok(res) => res,
+                Err(e) => {
+                    log::error!("Error getting nodes versions from peers: {}", e);
+                    Timer::sleep(10.).await;
+                    continue;
+                },
+            };
+
+            for (peer_id, response) in get_versions_res {
+                let name = match peers_names.get(&peer_id.to_string()) {
+                    Some(n) => n.clone(),
+                    None => continue,
+                };
+
+                match response {
+                    PeerDecodedResponse::Ok(version) => {
+                        let node_version_stat = NodeVersionStat {
+                            name,
+                            version,
+                            timestamp,
+                        };
+                        if let Err(e) = insert_node_version_stat_to_db(&ctx, &node_version_stat) {
+                            log::error!("Error inserting nodes versions into db: {}", e);
+                            continue;
+                        };
+                    },
+                    // If a node returns an error or no response it will not be added to the stats table
+                    // A simple count for every node in db will return a count for the number of responses recieved
+                    PeerDecodedResponse::Err(e) => {
+                        log::error!("Node {} responded to version request with error: {}", name, e);
+                        continue;
+                    },
+                    PeerDecodedResponse::None => {
+                        log::debug!("Node {} did not respond to version request", name);
+                        continue;
+                    },
+                }
+            }
+        }
+        Timer::sleep(interval).await;
+    }
+}
+
+fn addr_to_ipv4_string(address: &str) -> Result<String, String> {
+    let address_with_port = if address.contains(':') {
+        address.to_string()
+    } else {
+        format!("{}{}", address, ":0")
+    };
+    match address_with_port.as_str().to_socket_addrs() {
+        Ok(mut iter) => match iter.next() {
+            Some(addr) => {
+                if addr.is_ipv4() {
+                    Ok(addr.ip().to_string())
+                } else {
+                    ERR!("Address {} resolved to IPv6 {} which is not supported", address, addr)
```

Comment:
Please add the error enum wrapped into MmError from this fn. `ERR!` is deprecated along with `try_s!`.